### DISCUSSION TITLE
Exist a guide to train Speedy Speach or Vocoders?

### DISCUSSION THREAD

**Discussion by nicolalandro** [coqui-ai/TTS#598] [General Q&A]
I'm tryng to train Tacotron2 for italian voice, but the results for now is not so good. I want to try to use speedy speach to have result quickly but by trying the same configuration of train tacotron2 it crash. It is possible that problems depends also from vocoder? How can I train one?

**Comment by erogol** [comment]
Speedy Speech needs a trained Tacotron model to pre-compute alignments 

Try using GlowTTS or AlingTTS models. If you use the upcoming `TrainerAPI` branch, you can find more info for training models here https://tts.readthedocs.io/en/trainer-api/index.html

**Reply by nicolalandro** [reply]
Thank you, I just train a Tacotron2, and I'm try to train a glow_tts but I have an error of out of memory at epoch 2 I do not know why, but now I'm trying to create the dataset from CommonVoice so I need to be more speed to test it. I can also try to create my dataset if it is needed. My aim is to create a trained model for italian voice, have you have any advices?

**Reply by erogol** [reply]
OOM I'm VRAM or RAM ? 

There is Italian in the public MAILabs TTS dataset I believe.

**Reply by Sadam1195** [reply]
> Thank you, I just train a Tacotron2, and I'm try to train a glow_tts but I have an error of out of memory at epoch 2 I do not know why, but now I'm trying to create the dataset from CommonVoice so I need to be more speed to test it. I can also try to create my dataset if it is needed. My aim is to create a trained model for italian voice, have you have any advices?

try with lower batch size to avoid `out of memory error`
dataset you can play with. https://data.solak.de/data/Training/stt_tts/it_IT.tgz
@nicolalandro

**Reply by nicolalandro** [reply]
The error is CUDA error, so I reduce de batch size to 64 and after to 32 but the problem return. The first epoch finishes correctly, I do not understand why the second should fail.
This is the config file, and I do not set anything else outside
```
{
    "datasets": [
        {
            "name": "ljspeech",
            "path": "/media/mint/Barracuda/Datasets/CommonVoiceMozillaIta/cv-corpus-6.1-2020-12-11/it/",
            "meta_file_train": "voice_best_train.csv",
            "meta_file_val": "voice_best_val.csv"
        }
    ],
    "audio": {
        "fft_size": 1024,
        "win_length": 1024,
        "hop_length": 256,
        "frame_length_ms": null,
        "frame_shift_ms": null,
        "sample_rate": 44100,
        "preemphasis": 0.0,
        "ref_level_db": 20,
        "do_trim_silence": true,
        "trim_db": 60,
        "power": 1.5,
        "griffin_lim_iters": 60,
        "num_mels": 80,
        "mel_fmin": 50.0,
        "mel_fmax": 7600.0,
        "spec_gain": 1,
        "signal_norm": true,
        "min_level_db": -100,
        "symmetric_norm": true,
        "max_norm": 4.0,
        "clip_norm": true,
        "stats_path": "scale_stats_common.npy"
    },
    "gst":{
        "gst_embedding_dim": 256,
        "gst_num_heads": 4,
        "gst_num_style_tokens": 10
    },
    "model": "glow_tts",
    "run_name": "glow_tts",
    "run_description": "test.",
    "batch_size": 32,
    "eval_batch_size": 8,
    "num_loader_workers": 4,
    "num_val_loader_workers": 2,
    "text_cleaner": "phoneme_cleaners",
    "output_path": "/media/mint/Barracuda/Models/TTS/out_commonvoice",
    "phoneme_cache_path": "/media/mint/Barracuda/Models/TTS/out_commonvoice"
}
```
Thank you for the dataset information. I will try. Also this dataset has different voices (like common voice), I can train tacotron with different speakers? How to set this information in config file?

**Reply by Sadam1195** [reply]
> ```
> {
>     "datasets": [
>         {
>             "name": "ljspeech",
>             "path": "/media/mint/Barracuda/Datasets/CommonVoiceMozillaIta/cv-corpus-6.1-2020-12-11/it/",
>             "meta_file_train": "voice_best_train.csv",
>             "meta_file_val": "voice_best_val.csv"
>         }
>     ],
>     "audio": {
>         "fft_size": 1024,
>         "win_length": 1024,
>         "hop_length": 256,
>         "frame_length_ms": null,
>         "frame_shift_ms": null,
>         "sample_rate": 44100,
>         "preemphasis": 0.0,
>         "ref_level_db": 20,
>         "do_trim_silence": true,
>         "trim_db": 60,
>         "power": 1.5,
>         "griffin_lim_iters": 60,
>         "num_mels": 80,
>         "mel_fmin": 50.0,
>         "mel_fmax": 7600.0,
>         "spec_gain": 1,
>         "signal_norm": true,
>         "min_level_db": -100,
>         "symmetric_norm": true,
>         "max_norm": 4.0,
>         "clip_norm": true,
>         "stats_path": "scale_stats_common.npy"
>     },
>     "gst":{
>         "gst_embedding_dim": 256,
>         "gst_num_heads": 4,
>         "gst_num_style_tokens": 10
>     },
>     "model": "glow_tts",
>     "run_name": "glow_tts",
>     "run_description": "test.",
>     "batch_size": 32,
>     "eval_batch_size": 8,
>     "num_loader_workers": 4,
>     "num_val_loader_workers": 2,
>     "text_cleaner": "phoneme_cleaners",
>     "output_path": "/media/mint/Barracuda/Models/TTS/out_commonvoice",
>     "phoneme_cache_path": "/media/mint/Barracuda/Models/TTS/out_commonvoice"
> }
> ```

Try resample data and user 16k or 22050 sample rate.

**Reply by Sadam1195** [reply]
> Thank you for the dataset information. I will try. Also this dataset has different voices (like common voice), I can train tacotron with different speakers? How to set this information in config file?

I don't know much about that may be you need to use speaker embeddings for that. But @erogol  could better answer that.

**Comment by Sadam1195** [comment]
> It is possible that problems depends also from vocoder? How can I train one?

In my experience, yes vocoder also effects TTS models greatly so you need have to right configs for both TTS and vocoder model. If you want to train one, you can use new Trainer-API branch. It is much simpler than before once you get hold of everything.

**Comment by ireneb612** [comment]
@Sadam1195 were you able to train a model for the ItalianLanguage??

**Reply by nicolalandro** [reply]
I do not remember the exact link but I find that: http://www.openslr.org/94/ . Here exist an Italian dataset for TTS and in general it can be extracted from LibriVox. My problem now is the batch size that I cannot enlarge on my machine, so my trains are very bad, I think that with that dataset and some computational power it is possible also to train a tacotron.

**Reply by Sadam1195** [reply]
> @Sadam1195 were you able to train a model for the ItalianLanguage??

Yes, I was able to train it successfully. 
@ireneb612
